src/lib/mongoBackend/mongoRegistrationsGet.cpp

```diff
@@ -0,0 +1,317 @@
+{
+  std::vector<mongo::BSONElement> dbAttributeV = getFieldF(cr0, REG_ATTRS).Array();
+
+  for (unsigned int ix = 0; ix < dbAttributeV.size(); ++ix)
+  {
+    mongo::BSONObj  aobj     = dbAttributeV[ix].embeddedObject();
+    std::string     attrName = getStringFieldF(aobj, REG_ATTRS_NAME);
+
+    if (attrName != "")
+    {
+      regP->dataProvided.attributes.push_back(attrName);
+    }
+  }
+}
+
+
+
+/* ****************************************************************************
+*
+* setDataProvided -
+*
+* Make sure there is only ONE "contextRegistration" in the vector
+* If we have more than one, then the Registration is made in API V1 as this is not
+* possible in V2 and we cannot respond to the request using the current implementation of V2.
+* This function will be changed to work in a different way once issue #3044 is dealt with.
+*
+*/
+static bool setDataProvided(ngsiv2::Registration* regP, const mongo::BSONObj& r, bool arrayAllowed)
+{
+  std::vector<mongo::BSONElement> crV = getFieldF(r, REG_CONTEXT_REGISTRATION).Array();
+
+  if (crV.size() > 1)
+  {
+    return false;
+  }
+
+  //
+  // Extract the first (and only) CR from the contextRegistration vector
+  //
+  mongo::BSONObj cr0 = crV[0].embeddedObject();
+
+  setEntities(regP, cr0);
+  setAttributes(regP, cr0);
+  setProvider(regP, cr0);
+
+  return true;
+}
+
+
+
+/* ****************************************************************************
+*
+* setExpires -
+*/
+static void setExpires(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->expires = (r.hasField(REG_EXPIRATION))? getIntFieldF(r, REG_EXPIRATION) : -1;
+}
+
+
+
+/* ****************************************************************************
+*
+* setStatus -
+*/
+static void setStatus(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  regP->status = (r.hasField(REG_STATUS))? getStringFieldF(r, REG_STATUS): "";
+}
+
+
+
+/* ****************************************************************************
+*
+* setForwardingInformation -
+*/
+static void setForwardingInformation(ngsiv2::Registration* regP, const mongo::BSONObj& r)
+{
+  // No forwarding info until API V2 forwarding is implemented
+}
+
+
+
+/* ****************************************************************************
+*
+* mongoRegistrationsGet - 
+*/
+void mongoRegistrationsGet
+(
+  std::vector<ngsiv2::Registration>*  regV,
+  const std::string&                  tenant,
+  const std::vector<std::string>&     servicePathV,
+  OrionError*                         oeP
+)
+{
+  bool         reqSemTaken = false;
+  std::string  err;
+  mongo::OID   oid;
+  StatusCode   sc;
+
+  reqSemTake(__FUNCTION__, "Mongo Get Registrations", SemReadOp, &reqSemTaken);
+
+  LM_T(LmtMongo, ("Mongo Get Registrations"));
+
+  std::auto_ptr<mongo::DBClientCursor>  cursor;
+  mongo::BSONObj                        q;
+
+  q = BSON(REG_SERVICE_PATH << fillQueryServicePath(servicePathV));
+
```

Comment:
In addition, we need sorting in the query results. Use the same criteria than in subscriptions, i.e. by `_id`, which (due to how _id is built) means in chronological order.

```
q.sort(BSON("_id" << 1));
```

Reply:
Fixed in 72f4e9f